src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreTaskBase.cs

```diff
@@ -293,5 +374,21 @@ protected virtual void Dispose(bool disposing)
                 [nameof(Recursive)] = Recursive.ToString()
             };
         }
+
+        /// <summary>
+        /// Writes global properties to the specified stream.
+        /// </summary>
+        /// <param name="writer">The <see cref="BinaryWriter" /> to write the global properties to.</param>
+        /// <param name="globalProperties">A <see cref="Dictionary{TKey, TValue}" /> containing the global properties.</param>
+        internal static void WriteGlobalProperties(BinaryWriter writer, Dictionary<string, string> globalProperties)
+        {
+            writer.Write(globalProperties.Count);
+
+            foreach (KeyValuePair<string, string> option in globalProperties)
+            {
+                writer.Write(option.Key);
+                writer.Write(option.Value);
```

Comment:
Is it possible to for a property value to be `null`?  If yes, this will throw `ArgumentNullException`.

Reply:
Fortunately no, these values come from MSBuild and unset properties are `string.Empty`.